lib/Backends/BackendUtils.cpp

```diff
@@ -55,3 +55,105 @@ size_t glow::getValueOffset(
   assert(it != symbolTable.end() && "Symbol not found.");
   return it->second.offset;
 }
+
+runtime::RuntimeBundle
+glow::generateRuntimeBundle(const IRFunction *F,
+                            MemoryAllocator *constantAllocator,
+                            MemoryAllocator *placeholderAllocator,
+                            MemoryAllocator *activationsAllocator) {
+  // Handle Constants, Placeholders, and Activations, in that order.
+  /// Symbol table mapping symbol name to offset for runtime.
+  std::unordered_map<std::string, runtime::RuntimeSymbolInfo> symbolTable;
+  uint64_t constantMaxSize = 0;
+  uint64_t placeholderMaxSize = 0;
+  uint64_t activationsMaxSize = 0;
+
+  // Compute the offsets for Constants.
+  if (constantAllocator) {
+    for (auto &v : F->getGraph()->getParent()->getConstants()) {
+      assert(isa<WeightVar>(F->getWeightForNode(v)) && "Expected WeightVar");
+      auto *w = cast<WeightVar>(F->getWeightForNode(v));
+      auto numBytes = w->getSizeInBytes();
```

Comment:
I would expect this size to be backend specific (i.e., because of alignment constraints on the row and so on.)
Where do we account for that?

Reply:
The allocator has some logic as far as alignment goes. For anything more involved we would likely want to introduce a more featured allocator.

Reply:
Not something to fix here, more thoughts for food, why we split the different allocator per kind of data?
Basically I believe we would need something syncing the allocators to check that the whole thing fits, thus I wonder if we shouldn’t have just one allocator to start with.